Default histogram observer

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26516 Per-channel baseline
* #26520 control of observer/fake-quant operations
* **#26622 Default histogram observer**
* #26519 Serialization and range reduction support for Fake Quant/Observer
* #26876 Fake quantization enhancements for QAT/PTQ support- fix tests

Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

raghuramank10000 committed:
Default histogram observer



Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/observer.py:
```diff
@@ -586,6 +586,11 @@ def default_observer(**kwargs):
     kwargs.setdefault("reduce_range", True)
     return observer(MinMaxObserver, **kwargs)
 
+def default_l2_observer(**kwargs):
```
nit: should we use default_histogram_observer to align the names?

jerryzh168 commented on torch/quantization/observer.py:
```diff
@@ -586,6 +586,11 @@ def default_observer(**kwargs):
 
+def default_l2_observer(**kwargs):
+    # Restrict activations to be in the range (0,127)
+    print(kwargs)
```
remove

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

dzhulgakov commented on torch/quantization/observer.py:
```diff
@@ -605,8 +605,8 @@ def calculate_qparams(self):
 default_observer = MinMaxObserver.with_args(reduce_range=True)
 default_debug_observer = RecordingObserver
 default_weight_observer = MinMaxObserver.with_args(dtype=torch.qint8, qscheme=torch.per_tensor_symmetric)
+default_histogram_observer = HistogramObserver.with_args(reduce_range=True)
```
should it be "per architecture" defaults?

jerryzh168 replied:
do you mean having default_histogram_observer_fbgemm and default_histogram_observer_qnnpack?

raghuramank100 replied:
That comes in a later diff, where we specify an architecture to obtain a qconfig.
https://github.com/pytorch/pytorch/pull/26627

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
LG

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Default histogram observer"


Differential Revision: [D17508787](https://our.internmc.facebook.com/intern/diff/D17508787/)

[ghstack-poisoned]
